E1_TFHub_Sample_Deploy/export_test.py

```diff
@@ -0,0 +1,63 @@
+import tensorflow as tf
+import tensorflow_hub as hub
+import unittest
+import tempfile
+import shutil
+import os
+from absl.testing import absltest
+import export
+from export import train_and_export
+
+TMPDIR = tempfile.mkdtemp()
+model = None
+
+
+def load_model(path):
+  global model
+  if not model:
+    model = hub.load(path)
+  return model
+
+
+class TFHubMNISTTest(tf.test.TestCase):
+  def setUp(self):
+    file_ = tf.gather(tf.io.matching_files("test_data/*.jpg"), 0)
+    self.test_image = tf.expand_dims(
+        tf.image.decode_jpeg(
+            tf.io.read_file(file_)), 0)
+    self.test_label = int(str(file_.numpy()).split("/")[-1].split(".jpg")[0])
+
+  def test_model_exporting(self):
+    mocked_dict = export.sys.__dict__.copy()
+    devnull = open(os.devnull, "w")
+    mocked_dict["stdout"] = devnull
+    absltest.mock.patch.dict(
+        export.sys.__dict__,
+        mocked_dict,
+        clear=True).start()
+    train_and_export(epoch=1, export_path="%s/model/1" % TMPDIR)
+    self.assertTrue(os.listdir(TMPDIR))
+
+  def test_model_loadable(self):
+    if not os.path.exists("%s/model/1" % TMPDIR):
+      self.test_model_exporting()
+    model = load_model("%s/model/1" % TMPDIR)
+    output_ = model.call(
+        tf.zeros([1, 28, 28, 1], dtype=tf.uint8).numpy())
+    self.assertEqual(output_.shape, [1, 10])
+
+  def test_model_working(self):
```

Comment:
Maybe add test for empty input? Then call the tests `test_empty_input`, `test_sample_input`, etc. These test names appear in error logs (if there is a regression), so its good to have their names reflect what exactly is not working :-)

Reply:
Hey sachin, Empty inputs are checked. Using `tf.zeros`. is that what you are talking about?

Reply:
Aah. Didn't see that. Then rename `test_model_loadable` to `test_zero_input`, since its doing something beyond just simply loading the model.

Reply:
Done :)

Reply:
Rename this test to `test_sample_input`.

Reply:
Done :)